src/features/broker-protection/actions/fill-form.js

```diff
@@ -47,35 +26,80 @@ export function fillForm (action, userData) {
     return new SuccessResponse({ actionID: action.id, actionType: action.actionType, response: null })
 }
 
+/**
+ * Try to fill form elements. Collecting results + warnings for reporting.
+ * @param {HTMLElement} root
+ * @param {{selector: string; type: string}[]} elements
+ * @param {Record<string, any>} data
+ * @return {{
+ *     results: ({result: true} | {result: false; error: string})[],
+ *     warnings: {message:string}[]
+ * }}
+ */
+export function fillMany (root, elements, data) {
+    const results = []
+    const warnings = []
+
+    for (const element of elements) {
+        const inputElem = getElement(root, element.selector)
+        if (!inputElem) {
+            // todo: report this? it can occur when a selector is incorrect, but should we fail the entire action?
```

Comment:
I'm also wondering if we should fail here. In most cases, at least in the opt out form, we're only adding the required fields to the json, so if any of them aren't filled out then the form won't submit (and I think we'd want to know).

Reply:
@brianhall ok, if you agree then we can simply push an error here instead

Reply:
Yeah, let's do it. Thank you!